src/modules/ShopgateAppCodeExecutor.js

```diff
@@ -0,0 +1,62 @@
+import { executeWithRetry } from './executeWithRetry';
+import { isShopgateJSBridgeAvailable } from './isShopgateJSBridgeAvailable';
+
+/**
+ * This class takes care if Shopgate App related code should be executed or not
+ * depending on the environment
+ */
+export class ShopgateAppCodeExecutor {
+  /**
+   * @param {int} intervalInMiliseconds             interval in milisenconds
+   *                                                to check if a shopgate app environment is set
+   * @param {int} maximumIntervallTimeInMiliseconds maximum interval in miliseconds
+   *                                                to check if a shopgate app environment is set
+   */
+  constructor(intervalInMiliseconds = 25, maximumIntervallTimeInMiliseconds = 2500) {
+    this.intervalInMiliseconds = intervalInMiliseconds;
+    this.maximumIntervallTimeInMiliseconds = maximumIntervallTimeInMiliseconds;
+    this.callbacks = [];
+    this.evaluateShopgateApp();
+  }
+
+  /**
+   * Tries to execute a Shopgate App related code
+   *
+   * @param {function} callback the method to call if we are executed inside of a Shopgate App
+   */
+  execute(callback) {
+    if (!this.shopgateApp) {
+      this.callbacks.push(callback);
+      return;
+    }
+
+    callback();
+  }
+
+  /**
+   * Tries to identify if we are currently in a Shopgate App environment
+   * @private
+   */
+  evaluateShopgateApp() {
+    executeWithRetry(this.intervalInMiliseconds, this.maximumIntervallTimeInMiliseconds, () => {
+      if (!isShopgateJSBridgeAvailable()) {
+        return false;
+      }
+
+      this.shopgateApp = true;
+      this.executeQueuedCallbacks();
+
+      return true;
+    });
+  }
+
+  /**
+   * Will execute all saved callbacks
+   * @private
+   */
+  executeQueuedCallbacks() {
+    for (let i = 0; i < this.callbacks.length; i += 1) {
```

Comment:
forEach is the slowest of way to walk through an array therefore I won't use it. This code in specific is probably not performance related. Even this is the case I won't write inperformant code in such cases because the chance is high I will write forEach in performance relevant places, too.

Another perspective of code is the readablitiy. forEach is more readable but this for loop is also readable there I will leave it as it is.

I will add the check for the callback to be a function

Reply:
After discussion with Carsten I refactored it. First I tried "for (callback of this.callbacks) {  } but this is not allowed in AirBnb code style therefore I go for the forEach